assets/check

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+import boto3
+import botocore
+import json
+import re
+import sys
+from packaging import version as semver
+
+DEFAULT_AWS_REGION = "eu-west-2"
+DEFAULT_VER_REGEX = "(\d{1,3}\.\d{1,3}\.\d{1,3})$"
+REQ_PARAMS = [
+    "aws_access_key_id",
+    "aws_secret_access_key",
+    "aws_region",
+    "ami_prefix",
+    "version_regex"
+]
+
+
+def validate_source(payload):
+    if "aws_region" not in payload['source'].keys():
+        payload['source'].update({"aws_region": DEFAULT_AWS_REGION})
+
+    if "version_regex" not in payload["source"].keys():
+        payload["source"].update({"version_regex": DEFAULT_VER_REGEX})
+
+    for param in REQ_PARAMS:
+        if param not in payload['source'].keys():
+            sys.stderr.write("\nRequired parameter {} not defined\n".format(param))
+            sys.exit(1)
+
+    return payload
+
+
+def return_versions(payload):
+    try:
+        session = boto3.Session(
+            aws_access_key_id=payload['source']['aws_access_key_id'],
+            aws_secret_access_key=payload['source']['aws_secret_access_key'],
+            region_name=payload['source']['aws_region']
+        )
+        ec2_client = session.client(
+            "ec2",
+            payload['source']['aws_region']
+        )
+
+        response = ec2_client.describe_images(
+            Filters=[
+                {
+                    'Name': 'name',
+                    'Values': [payload['source']['ami_prefix'] + '*']
+                },
+            ],
+            Owners=['self']
```

Comment:
Am I right in thinking this will limit results only to the current account? We're definitely going to need to pull images across accounts

Reply:
You're correct. I've updated this to be the default but can now ben optionally overridden via the `aws_owner_ids` source parameter.